src/tracks/playabletrack/wavetrack/ui/LowlitClipButton.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "ClipButtonId.h"
+#include "UIHandle.h"
+#include "WaveTrack.h"
+
+class wxRect;
+class wxPoint;
+
+template <ClipButtonId id> struct ClipButtonTag
+{
+};
+
+struct ClipButtonDrawingArgs
+{
+   const wxRect& rect;
+   const ClipInterface& clip;
+   wxDC& dc;
+};
+
+namespace ClipButtonSpecializations
+{
+bool NeedsDrawing(ClipButtonTag<ClipButtonId::Overflow>, const ClipInterface&);
```

Comment:
You could avoid introducing extra name `ClipButtonTag`:
```
template<ClipButtonId ID>
class ClipButtonSpecializations final
{
public:
   static bool NeedsDrawing(const ClipInterface&);
   ...
};
extern template<> ClipButtonSpecializations<ClipButtonId::Overflow>;
...
```

Reply:
Nice!